qutebrowser/utils/urlmatch.py

```diff
@@ -177,6 +179,15 @@ def _init_host(self, parsed):
             assert self._host is None
             return
 
+        if parsed.netloc.startswith('['):
+            # Using QUrl parsing to minimize ipv6 addresses
+            url = QUrl()
+            url.setHost("[" + parsed.hostname + "]")
```

Comment:
I don't think the brackets are required with `setHost`?

Reply:
This is needed otherwise if the hostname is `[::1]` it isn't declined

Reply:
ok, I read the python bug report you wrote. It seems assuming `urllib.parse.urlparse` works correctly we don't need the brackets